### PR TITLE
Add Zsh completion

### LINKED ISSUE #54: Shell completion
It would be nice if `comma` shipped with shell completion files (i.e: make it so that the shell completes as if `,` was not there).

### PR DESCRIPTION
**Pull request by ambroisie**
Closes #54 (only for Zsh, as that's the one I use).

### CODE REVIEW

File: contrib/zsh/_comma [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+local -a args
+
+args=(
+  # Comma options
+  '(- 1 *)'{-h,--help}'[display usage information]'
+  '(- 1 *)'{-V,--version}"[show program's version number and exit]"
+  '(-i --install)'{-i,--install}'[install the derivation containing the executable]'
+  '(-s --shell)'{-s,--shell}'[open a shell containing the derivation containing the executable]'
+  '--picker[which picker to use when selecting a derivation]:<PICKER> [fzy]: _command_names -e'
+  '--nixpkgs_flake[which nixpkgs flake to use when selecting a derivation]:<NIXPKGS_FLAKE> [nixpkgs]'
```
Comment (ambroisie) [inline contrib/zsh/_comma:12]: One missing feature would be to do something similar to [`nix-zsh-completions`](https://github.com/nix-community/nix-zsh-completions) and have suggestions of all flakes in the registry.

**Comment by Artturin** [comment]
Using https://docs.rs/clap_complete/latest/clap_complete/ would be nicer and would support more shells (bash,elvish,fish,powershell,zsh)

https://github.com/crowlKats/clap/blob/master/clap_complete/examples/completion-derive.rs

**Comment by ambroisie** [comment]
Yeah but does it know how to auto-complete for sudo-like command wrappers? I couldn't find it in the docs.

**Comment by Artturin** [comment]
Maybe
https://docs.rs/clap/latest/clap/enum.ValueHint.html#variant.CommandWithArguments

https://github.com/clap-rs/clap/blob/14fd853fb9c5b94e371170bbd0ca2bf28ef3abff/clap_complete/examples/completion-derive.rs#L56-L58

https://github.com/search?q=%2FValueHint%3A%3ACommandWithArguments%2F+lang%3Arust+NOT+is%3Afork&type=code

https://github.com/clap-rs/clap/blob/14fd853fb9c5b94e371170bbd0ca2bf28ef3abff/clap_complete/src/shells/zsh.rs#L408

https://github.com/clap-rs/clap/blob/14fd853fb9c5b94e371170bbd0ca2bf28ef3abff/clap_complete/tests/snapshots/value_hint.zsh#L40

**Comment by ambroisie** [comment]
Ah nice, I was looking for `_command_names`.

**Pull request closed without merging**